kernel/cClosure.ml

```diff
@@ -301,33 +307,34 @@ let destFLambda clos_fun t =
   | FLambda(_,[(na,ty)],b,e) ->
     (usubst_binder e na,clos_fun e ty,clos_fun (usubs_lift e) b)
   | FLambda(n,(na,ty)::tys,b,e) ->
-    (usubst_binder e na,clos_fun e ty,{mark=t.mark;term=FLambda(n-1,tys,b,usubs_lift e)})
+    (usubst_binder e na,clos_fun e ty,{mark=t.mark;term=FLambda(n-1,tys,b,usubs_lift e); as_constr=None})
   | _ -> assert false
 
 (* Optimization: do not enclose variables in a closure.
    Makes variable access much faster *)
+(* XXX some of these as_constr could be "Some t" *)
```

Comment:
I think my vague proposal boils down to the same thing as this comment here, in `mk_clos` and likely in all other places where `FInd`, `FConstruct`, `FFlex` (maybe not all cases of that?), `FAtom`, and primitive nodes such as `FFloat`, `FInt`, and `FString` are turned into `fconstr`. We should be able to get away with `Some (ref (Closed <original term>))` for most of these, right?